wetterdienst/provider/dwd/observation/parser.py

```diff
@@ -209,18 +182,45 @@ def _parse_climate_observations_data(
         for parameter in DwdObservationParameter[resolution.name][dataset.name]:
             columns.append(parameter.value)
 
-        df = df.reindex(columns=columns)
+        df = df.with_columns(columns)
 
     # Special handling for hourly solar data, as it has more date columns
     if resolution == Resolution.HOURLY:
         if dataset == DwdObservationDataset.SOLAR:
             # Fix real date column by cutting of minutes
-            df["mess_datum"] = df["mess_datum"].str[:-3]
+            df = df.with_columns(pl.col("mess_datum").str.slice(0, pl.col("mess_datum").str.n_chars() - 3))
 
     if resolution in (Resolution.MONTHLY, Resolution.ANNUAL):
-        df = df.drop(columns=["bis_datum", "mess_datum_ende"], errors="ignore").rename(
-            columns={"mess_datum_beginn": Columns.DATE.value}
+        df = df.drop(columns=[col for col in ["bis_datum", "mess_datum_ende"] if col in df.columns]).rename(
+            mapping={"mess_datum_beginn": "mess_datum"}
+        )
+
+    if resolution == Resolution.SUBDAILY and dataset == DwdObservationDataset.WIND_EXTREME:
```

Comment:
Aren't identity checks the right way to compare Enum items?
```suggestion
    if resolution is Resolution.SUBDAILY and dataset is DwdObservationDataset.WIND_EXTREME:
```